### ISSUE TITLE
Cannot use fasttext in centos 7

### ISSUE THREAD

**Issue by ajan1019** [facebookresearch/fastText#809]
I built fasttext from the source. When I tried to use in centos 7, I got below error.

`/usr/lib64/libstdc++.so.6: version 'GLIBCXX_3.4.20' not found (required by /workspace/azar/lib/python3.6/site-packages/fasttext_pybind.cpython-36m-x86_64-linux-gnu.so)`

**Machine details:**
OS-version: Centos 7

I dug more into the web, But couldn't come up with a way to solve this issue.

I tried various methods to make this works, But unfortunately, all ended in a negative result.

Some of the methods are,

1) installed the latest version of GCC using devtoolset.
  - It didn't work because even I installed latest GCC still it takes libstdc++.so.6 from system default location which /usr/lib64/libstdc++.so.6.

When I run strings  `/usr/lib64/libstdc++.so.6 | grep GLIBCXX`, Below is the output.

> GLIBCXX_3.4
> GLIBCXX_3.4.1
> GLIBCXX_3.4.2
> GLIBCXX_3.4.3
> GLIBCXX_3.4.4
> GLIBCXX_3.4.5
> GLIBCXX_3.4.6
> GLIBCXX_3.4.7
> GLIBCXX_3.4.8
> GLIBCXX_3.4.9
> GLIBCXX_3.4.10
> GLIBCXX_3.4.11
> GLIBCXX_3.4.12
> GLIBCXX_3.4.13
> GLIBCXX_3.4.14
> GLIBCXX_3.4.15
> GLIBCXX_3.4.16
> GLIBCXX_3.4.17
> GLIBCXX_3.4.18
> GLIBCXX_3.4.19
> GLIBCXX_DEBUG_MESSAGE_LENGTH

It looks like it is the known issue with centos 7. I don't have this problem in either mac os or Ubuntu. 

Please see the below link to know more about this issue in centos.

[centos issue reference](https://stackoverflow.com/questions/52609269/c-project-compiled-with-modern-compiler-but-linked-against-outdated-libstdc)

**Comment by ajan1019** [comment]
Found what was the problem and fixed it.

Problem is I am using both pytorch and fasttext in my repo.
Pytorch has been compiled using gcc-4.9.2.
Initially, I got segmentation fault error when I compiled using python3.6(with gcc 4.8.5).
To know more about that please refer [this
](https://github.com/pytorch/pytorch/issues/9129).
To avoid this error I compiled fasttext using ubuntu which has GLIBCXX_3.4.20. 
When I tried to install that in Centos 7 I got above error. 

The solution is to compile using a centos-7 machine which has GCC above 5.
Make sure python in which we are going to compile is having GCC version above 5. This can be achieved by compiling Python from the source.

**Why this issue comes only in centos?**
GLIBC depends on GCC version. But, Default GCC for centos is 4.8.5. So, when you upgrade your GCC using development tools, it tries to make sure it works for all GCC version. So, what centos does is, it will use basic libstdc++.so.6 which has only up to GLIBCXX_3.4.19. By doing this it makes sure it will work even on machines which has GCC version 4.8.5.

**TLDR: When I compiled using centos 7 & python3.6(with GCC 8) problem got solved.**
